docs/tutorial-node-implementation.md

```diff
@@ -0,0 +1,497 @@
+        return
+    extract_zip(zip_da, Path('plugins'))
+
+
+def post_install(environ):
+    extra_env = {
+        'FLASK_APP': 'qhana_plugin_runner',
+        'FLASK_ENV': 'production',
+        'PLUGIN_FOLDERS': 'plugins:git-plugins'
+    }
+
+    cmd = ['python3', '-m', 'invoke', 'load-git-plugins']
+    print(cmd_join(cmd), 'env:', extra_env)
+    subprocess.run(cmd, env=ChainMap(extra_env, environ))
+    cmd = ['python3', '-m', 'flask', 'install']
+    print(cmd_join(cmd), 'env:', extra_env)
+    subprocess.run(cmd, env=ChainMap(extra_env, environ))
+
+
+def main(argv):
+    print('pwd', Path('.').resolve())
+    print(argv)
+    env = ChainMap(argv_to_dict(argv), environ)
+    install_requirements(env)
+    install_deployment_artifacts(env)
+    post_install(env)
+
+
+if __name__ == "__main__":
+    main(sys.argv[1:])
+```
+
+### Explanation
+
+`PythonScriptArtifacts` get the user input not as environment variables but as command-line arguments.
+The function `argv_to_dict` converts the command-line arguments to a dictionary for easy access.
+The `ChainMap` class is used to merge multiple dictionaries, i.e., to merge the command-line arguments with the environment variables.
+`install_requirements` installs git.
+The file names of the `Deployment Artifacts` are passed as environment variables to the script under the key `DAs`, and the path to the unzipped `CSAR` folder can be found under the key `CSAR`.
+The function `install_deployment_artifacts` calculates the paths to the `Deployment Artifacts`,
+checks the type, and calls the right install functions.
+`install_plugin_runner` unpacks the passed `Deployment Artifact` and installs required Python packages, `install_plugin` unpacks the passed `Deployment Artifact` into the plugins folder.
+In the end, `post_install` executes commands to configure the application and install more dependencies.
+
+
+## Deployment Artifacts
+
+`Deployment Artifacts` implement the business logic of an application, e.g., the jar file for a Tomcat web server.
+If you want to create a `Node Type` that has, e.g., a jar file as a deployment artifact, you need to create an interface with an implementation artifact that can execute this file.
+
+**Example with a jar file as deployment artifact**
+- create a `Node Type`
+- create an `Interface` with a `start operation` for this `Node Type`
+- create a `Node Type Implementation` for this `Node Type`
+- add a `Deployment Artifact` to the `Node Type Implementation`
+![deployment artifact](./images/new_node_type/deployment%20artifacts/jar_artifact.png)
+- upload the `jar file` to the `Deployment Artifact`
+- add a `ScriptArtifact` as `Implementation Artifact` for the `start operation` to the `Node Type Implementation`
+![implementation artifact](./images/new_node_type/deployment%20artifacts/start_artifact.png)
+- upload a `script` to the `ScriptArtifact` that executes the `jar file`
+
+Example script to execute a `jar file`:
+```bash
```

Comment:
Maybe instead of only starting the JAR, add some exmple that finds the DAs current location using the $DAs variable and moves it somewhere first. [Exmaple](https://github.com/OpenTOSCA/tosca-definitions-common/blob/main/artifacttemplates/http%253A%252F%252Fopentosca.org%252Fartifacttemplates/SpringWebApp_Install-w1/files/install.sh)